Tasks/CircleCIDownloadArtifactsV0/download.ts

```diff
@@ -0,0 +1,128 @@
+
+function getDefaultProps() {
+    var hostType = (tl.getVariable('SYSTEM.HOSTTYPE') || "").toLowerCase();
+    return {
+        hostType: hostType,
+        definitionName: '[NonEmail:' + (hostType === 'release' ? tl.getVariable('RELEASE.DEFINITIONNAME') : tl.getVariable('BUILD.DEFINITIONNAME')) + ']',
+        processId: hostType === 'release' ? tl.getVariable('RELEASE.RELEASEID') : tl.getVariable('BUILD.BUILDID'),
+        processUrl: hostType === 'release' ? tl.getVariable('RELEASE.RELEASEWEBURL') : (tl.getVariable('SYSTEM.TEAMFOUNDATIONSERVERURI') + tl.getVariable('SYSTEM.TEAMPROJECT') + '/_build?buildId=' + tl.getVariable('BUILD.BUILDID')),
+        taskDisplayName: tl.getVariable('TASK.DISPLAYNAME'),
+        jobid: tl.getVariable('SYSTEM.JOBID'),
+        agentVersion: tl.getVariable('AGENT.VERSION'),
+        agentOS: tl.getVariable('AGENT.OS'),
+        agentName: tl.getVariable('AGENT.NAME'),
+        version: taskJson.version
+    };
+}
+
+function publishEvent(feature, properties: any): void {
+    try {
+        var splitVersion = (process.env.AGENT_VERSION || '').split('.');
+        var major = parseInt(splitVersion[0] || '0');
+        var minor = parseInt(splitVersion[1] || '0');
+        let telemetry = '';
+        if (major > 2 || (major == 2 && minor >= 120)) {
+            telemetry = `##vso[telemetry.publish area=${area};feature=${feature}]${JSON.stringify(Object.assign(getDefaultProps(), properties))}`;
+        }
+        else {
+            if (feature === 'reliability') {
+                let reliabilityData = properties;
+                telemetry = "##vso[task.logissue type=error;code=" + reliabilityData.issueType + ";agentVersion=" + tl.getVariable('Agent.Version') + ";taskId=" + area + "-" + JSON.stringify(taskJson.version) + ";]" + reliabilityData.errorMessage
+            }
+        }
+        console.log(telemetry);;
+    }
+    catch (err) {
+        tl.warning("Failed to log telemetry, error: " + err);
+    }
+}
+
+async function main(): Promise<void> {
+    var promise = new Promise<void>(async (resolve, reject) => {
+        let connection = tl.getInput("connection", true);
+        let definitionId = tl.getInput("definition", true);
+        let buildId = tl.getInput("version", true);
+        let itemPattern = tl.getInput("itemPattern", false);
+        let downloadPath = tl.getInput("downloadPath", true);
+        var retryLimit = parseInt(tl.getVariable("VSTS_HTTP_RETRY")) ? parseInt(tl.getVariable("VSTS_HTTP_RETRY")) : 4;
+
+        var endpointUrl = tl.getEndpointUrl(connection, false);
+        var itemsUrl = endpointUrl + "/api/v1.1/project/"+definitionId+"/"+buildId+"/artifacts";
+        itemsUrl = itemsUrl.replace(/([^:]\/)\/+/g, "$1");
+        console.log(tl.loc("DownloadArtifacts", itemsUrl));
+
+        var templatePath = path.join(__dirname, 'circleCI.handlebars.txt');
+        var username = tl.getEndpointAuthorizationParameter(connection, 'username', false);
```

Comment:
username isn't treated as secret in UI, how do we handle not exposing these secrets to users with view endpoint permission?

Reply:
added isconfidential prop for username field in the endpoint